app/src/componentLibrary/dropdown/utils.js

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2022 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+export const calculateDefaultIndex = (options, selectedValue) =>
+  options
+    .map(({ value }) => value)
+    .indexOf(selectedValue instanceof Object ? selectedValue.value : selectedValue);
+
+const calculateCurrentItemIndex = (index, itemsCount) =>
+  ((index % itemsCount) + itemsCount) % itemsCount;
+
+export const calculatePrevIndex = (index, options) => {
+  if (!options[index].disabled) {
+    return index;
+  }
+  const itemsCount = options.length;
+
+  return calculatePrevIndex(calculateCurrentItemIndex(index - 1, itemsCount), options);
```

Comment:
Are you calling this function recursively to skip disabled options?
Could you please replace it with a cycle?

Reply:
Well, let's leave it as is.
But one more note: 
these two functions looks too similar, can we unit them together and reuse it via more specific ones?
f.e.
```
export const findNearestAvailableIndex = (index, options, step = 1) => {
 if (!options[index].disabled) {
    return index;
  }
 const itemsCount = options.length;

 return findNextAvailableIndex(calculateCurrentItemIndex(index + step, itemsCount), options);
}

export const calculateNextIndex = (index, options) => findNearestAvailableIndex(index, options);

export const calculatePrevIndex = (index, options) => findNearestAvailableIndex(index, options, -1);
```